Documentation/CodingGuidelines/CodingBestPractices/NamedArguments.rst

```diff
@@ -0,0 +1,247 @@
+the Continuous Integration (CI) pipeline to ensure consistency and integrity over time.
+
+It’s important to note that the TYPO3 core team will not accept patches that aim
+to unilaterally transition the codebase from positional arguments to named arguments
+or vice versa without clear further benefits.
+
+Leveraging Named Arguments in PCPP Value Objects
+------------------------------------------------
+
+Recent advancements in the TYPO3 core codebase emphasize the separation of
+functionality and state, leading to the broad utilisation of value objects.
+Consider the following example:
+
+..  code-block:: php
+    :caption: Value object using public constructor property promotion
+
+    final readonly class Label implements \JsonSerializable
+    {
+        public function __construct(
+            public string $label,
+            public string $color = '#ff8700',
+            public int $priority = 0,
+        ) {}
+
+        public function jsonSerialize(): array
+        {
+            return get_object_vars($this);
+        }
+    }
+
+Using public constructor property promotions (PCPP) facilitates object
+initialization, representing one of the primary use cases for named arguments
+envisioned by PHP developers:
+
+.. code-block:: php
+    :caption: Instantiate a PCPP value object using named arguments
+
+    $label = new Label(
+        label: $myLabel,
+        color: $myColor,
+        priority: -1,
+    );
+
+Objects with such class signatures MUST be instantiated using named arguments to
+maintain API consistency. Standardizing named argument usage allows the TYPO3
+core to introduce deprecations for argument removals seamlessly.
+
+Invoking 2nd Party Dependency Methods
+-------------------------------------
+
+The TYPO3 core refrains from employing named arguments when calling library
+code from dependent packages unless the library explicitly mandates such usage
+and defines its variable names as part of the API, a practice seldom observed
+currently.
+
+As package consumer, the TYPO3 core must assume that packages don’t treat their
+variable names as API, they may change anytime. If TYPO3 core would use named
+arguments for library calls, this may trigger regressions: Suppose a patch level
+release of a library changes a variable name of some method that we call using
+named arguments. This would immediately break when TYPO3 projects upgrade to
+this patch level release due to the power of semantic versioning. TYPO3 core must
+avoid this scenario.
+
+Invoking core API
+-----------------
+
+Within the TYPO3 core, named arguments are not used when invoking its own methods.
+There are exceptions in specific scenarios as outlined below, however these are
+the reasons for not using named arguments:
+
+* TYPO3 core tries to be as consistent as possible
+
+* Setting a good example for extensions
+
+* Avoiding complications and side effects during refactoring
```

Comment:
```suggestion
* Avoiding complications and side effects during refactoring, also preventing accidental semantic versioning breaking changes in patch or feature level releases
```

Or maybe something like "prevent introducing a layer of public API responsibility"?

Reply:
not the intention here. 'preventing accidental semantic versioning breaking' is ruled out anyway. this is about changing a variable name of a method and then being forced to adapt all consumers as well.